tests/core/string/test_string.h

```diff
@@ -469,6 +471,28 @@ TEST_CASE("[String] String to integer") {
 	CHECK(String("999999999999999999999999999999999999999999999999999999999").to_int() == INT64_MAX); // Too large, largest possible is returned.
 	CHECK(String("-999999999999999999999999999999999999999999999999999999999").to_int() == INT64_MIN); // Too small, smallest possible is returned.
 	ERR_PRINT_ON
+
+	// This method stops at the first non-digit character.
+
+	static const char *nums_char[14] = { "1237461283", "-22", "0", "-1123412", "", "10_000_000", "-1_2_3_4", "-9223372036854775808", "+12.34 ", "-0", "007", "-+45", "-46/46/46", "-7-2" };
+	static const wchar_t *nums_wchar[14] = { L"1237461283", L"-22", L"0", L"-1123412", L"", L"10_000_000", L"-1_2_3_4", L"-9223372036854775808", L"+12.34 ", L"-0", L"007", L"-+45", L"-46/46/46", L"-7-2" };
+	static const char32_t *nums_char32[14] = { U"1237461283", U"-22", U"0", U"-1123412", U"", U"10_000_000", U"-1_2_3_4", U"-9223372036854775808", U"+12.34 ", U"-0", U"007", U"-+45", U"-46/46/46", U"-7-2" };
+	static const int64_t integers[14] = { 1237461283, -22, 0, -1123412, 0, 10, -1, -9223372036854775808LL, 12, 0, 7, 0, -46, -7 };
```

Comment:
```suggestion
	static const int64_t integers[14] = { 1237461283, -22, 0, -1123412, 0, 10, -1, -9223372036854775808uLL, 12, 0, 7, 0, -46, -7 };
```
Possible that the negation is unary and not part of the constant, but strange

Reply:
> Possible that the negation is unary and not part of the constant, but strange

i think should be (-9223372036854775807LL -1) as it contains literals that are within the range.